src/main/java/io/github/opencubicchunks/cubicchunks/cubicgen/customcubic/CustomGeneratorSettings.java

```diff
@@ -301,237 +360,198 @@ public static CustomGeneratorSettings fromVanilla(ChunkGeneratorSettings setting
+                    {
+                        JsonObject properties = new JsonObject();
+                        properties.add("level", new JsonPrimitive("0"));
+                        oceanblock.add("Properties", properties);
+                        oceanblock.add("Name", new JsonPrimitive("minecraft:water"));
+                    }
+                    defaults.add(MODID + ":biome_fill_depth_offset", new JsonPrimitive(3.0f));
+                    defaults.add(MODID + ":biome_fill_noise_octaves", new JsonPrimitive(4.0f));
+                    defaults.add(MODID + ":height_scale", new JsonPrimitive(factor));
+                    defaults.add(MODID + ":biome_fill_depth_factor", new JsonPrimitive(2.3333333333333335f));
+                    defaults.add(MODID + ":mesa_depth", new JsonPrimitive(16.0f));
+                    defaults.add(MODID + ":water_level", root.get("waterLevel"));
+                    defaults.add(MODID + ":biome_fill_noise_freq", new JsonPrimitive(0.0078125f));
                 }
-                compound.setString("generatorOptions", gson.toJson(root));
-                return compound;
+                replacerConf.add("defaults", defaults);
+                replacerConf.add("overrides", new JsonObject());
             }
-        });
-
 
-        fixes.registerFix(FixTypes.LEVEL, new IFixableData() {
-            @Override public int getFixVersion() {
-                return 2;
-            }
+            root.add("replacerConfig", replacerConf);
+        }
+        return gson.toJson(root);
+    }
 
-            @Override public NBTTagCompound fixTagCompound(NBTTagCompound compound) {
-                if (!compound.getString("generatorName").equals("CustomCubic")) {
-                    return compound;
-                }
-                String generatorOptions = compound.getString("generatorOptions");
-                if (generatorOptions.isEmpty()) {
-                    generatorOptions = new CustomGeneratorSettings().toJson(true);
-                }
-                // this is far simpler that walking through the json and figurring out all the places where it occurs
-                // instead, just do string search and replace. The string shouldn't occur in any other context
-                compound.setString("generatorOptions", generatorOptions.replaceAll(MODID + ":", MODID + ":"));
-                return compound;
-            }
-        });
+    public static String fixGeneratorOptionsVersion3(String generatorOptions) {
+        if (generatorOptions.isEmpty()) {
+            generatorOptions = new CustomGeneratorSettings().toJson(true);
+        }
+        // this is far simpler that walking through the json and figurring out
+        // all the places where it occurs
+        // instead, just do string search and replace. The string shouldn't
+        // occur in any other context
+        return generatorOptions.replaceAll(MODID + ":", MODID + ":");
```

Comment:
Uh... ok, I didn't notice that it's broken, it should replace cubicchunks with cubicgen. Can you fix that too?